src/datasources/email/email.datasource.spec.ts

```diff
@@ -440,7 +466,255 @@ describe('Email Datasource Tests', () => {
+              VALUES (${subscriptions[0].key}, ${subscriptions[0].name}),
+                     (${subscriptions[1].key}, ${subscriptions[1].name})`;
+    await target.saveEmail({
+      chainId,
+      safeAddress,
+      emailAddress,
+      account,
+      code,
+      codeGenerationDate,
+      unsubscriptionToken,
+    });
+    // Subscribe to two categories
+    await target.subscribe({
+      chainId,
+      safeAddress,
+      account,
+      categoryKey: subscriptions[0].key,
+    });
+    await target.subscribe({
+      chainId,
+      safeAddress,
+      account,
+      categoryKey: subscriptions[1].key,
+    });
+
+    // Unsubscribe from one category
+    const result = await target.unsubscribe({
+      categoryKey: subscriptions[0].key,
+      token: unsubscriptionToken,
+    });
+
+    const currentSubscriptions = await target.getSubscriptions({
+      chainId,
+      safeAddress,
+      account,
+    });
+    expect(result).toContainEqual(subscriptions[0]);
+    expect(currentSubscriptions).not.toContainEqual(subscriptions[0]);
+  });
+
+  it('unsubscribes to non existent category', async () => {
+    const chainId = faker.number.int({ max: DB_CHAIN_ID_MAX_VALUE }).toString();
+    const safeAddress = faker.finance.ethereumAddress();
+    const emailAddress = new EmailAddress(faker.internet.email());
+    const account = faker.finance.ethereumAddress();
+    const code = faker.string.numeric();
+    const codeGenerationDate = faker.date.recent();
+    const unsubscriptionToken = faker.string.uuid();
+    const nonExistentCategory = faker.word.sample();
+    await target.saveEmail({
+      chainId,
+      safeAddress,
+      emailAddress,
+      account,
+      code,
+      codeGenerationDate,
+      unsubscriptionToken,
+    });
+
+    // Unsubscribe from one category
+    const result = await target.unsubscribe({
+      categoryKey: nonExistentCategory,
+      token: unsubscriptionToken,
+    });
+
+    expect(result).toHaveLength(0);
+  });
+
+  it('unsubscribes with wrong token', async () => {
+    const chainId = faker.number.int({ max: DB_CHAIN_ID_MAX_VALUE }).toString();
+    const safeAddress = faker.finance.ethereumAddress();
+    const emailAddress = new EmailAddress(faker.internet.email());
+    const account = faker.finance.ethereumAddress();
+    const code = faker.string.numeric();
+    const codeGenerationDate = faker.date.recent();
+    const unsubscriptionToken = faker.string.uuid();
+    const subscriptions = [
+      {
+        key: faker.word.sample(),
+        name: faker.word.words(2),
+      },
+    ];
+    await target.saveEmail({
+      chainId,
+      safeAddress,
+      emailAddress,
+      account,
+      code,
+      codeGenerationDate,
+      unsubscriptionToken,
+    });
+
+    // Unsubscribe from one category
+    const result = await target.unsubscribe({
+      categoryKey: subscriptions[0].key,
+      token: faker.string.uuid(),
+    });
+
+    expect(result).toHaveLength(0);
```

Comment:
I think we should also check that the subscriptions of `emailAddress` has not been touched after this.

Reply:
Addressed in [040d7d6](https://github.com/safe-global/safe-client-gateway/pull/1028/commits/040d7d6378f9fe26daf364d342d3a9481c361b8a)